src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IBlock.cs

```diff
@@ -0,0 +1,514 @@
+
+            compilation.VerifyDiagnostics();
+
+            string expectedGraph = @"
+Block[0] - Entry
+    Statements (0)
+    Next (Regular) Block[1]
+Block[1] - Exit
+    Predecessors: [0]
+    Statements (0)
+";
+            VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedGraph);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void BlockFlow_03()
+        {
+            var source = @"
+#pragma warning disable CS0219
+
+class C
+{
+    void F()
+    /*<bind>*/{
+        int i;
+        i = 1;
+        {
+            int j;
+            j = 1;
+        }
+    }/*</bind>*/
+}";
+
+            var compilation = CreateStandardCompilation(source, parseOptions: TestOptions.RegularWithFlowAnalysisFeature);
+
+            compilation.VerifyDiagnostics();
+
+            string expectedGraph = @"
+Block[0] - Entry
+    Statements (0)
+    Next (Regular) Block[1]
+        Entering: {1}
+
+.locals {1}
+{
+    Locals: [System.Int32 i]
+    Block[1] - Block
+        Predecessors: [0]
+        Statements (1)
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'i = 1;')
+              Expression: 
+                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'i = 1')
+                  Left: 
+                    ILocalReferenceOperation: i (OperationKind.LocalReference, Type: System.Int32) (Syntax: 'i')
+                  Right: 
+                    ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+
+        Next (Regular) Block[2]
+            Entering: {2}
+
+    .locals {2}
+    {
+        Locals: [System.Int32 j]
+        Block[2] - Block
+            Predecessors: [1]
+            Statements (1)
+                IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'j = 1;')
+                  Expression: 
+                    ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'j = 1')
+                      Left: 
+                        ILocalReferenceOperation: j (OperationKind.LocalReference, Type: System.Int32) (Syntax: 'j')
+                      Right: 
+                        ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+
+            Next (Regular) Block[3]
+                Leaving: {2} {1}
+    }
+}
+
+Block[3] - Exit
+    Predecessors: [2]
+    Statements (0)
+";
+            VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedGraph);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void BlockFlow_04()
+        {
+            var source = @"
+#pragma warning disable CS0219
+
+class C
+{
+    void F()
+    /*<bind>*/{
+        int i;
+        {
+            int j;
+            i = 1;
```

Comment:
Now, this is what I mean by documenting what lifetime means. The fact that there is only one region in this function feels very surprising to me.

Reply:
Can you add a test similar to this, except that `i` is then used outside this block?

---
In reply to: [167707849](https://github.com/dotnet/roslyn/pull/24770#discussion_r167707849) [](ancestors = 167707849)